### ISSUE TITLE
Rule Request: identifier_name customization allow 'id' when type is conforming to Identifiable protocol

### ISSUE THREAD

**Issue by edorphy** [realm/SwiftLint#3651]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Swift has introduced a very commonly used protocol named Identifiable. It is used all over the place in SwiftUI. Out of the box, this `id` should be allowed when type conforms to the protocol. As a user of the library, I don't want to have to have to configure the exclusion list for this rule to include `id` because that could potentially allow real issues to slip through.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Shouldn't trigger

```
struct TaproomAnnotation: Identifiable {
    let id: UUID = UUID()
    let coordinate: CLLocationCoordinate2D
}
```

Should trigger
```
struct TaproomAnnotation {
    let id: UUID = UUID()
    let coordinate: CLLocationCoordinate2D
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Yes, this is a proposal to allow a configuration point for the existing identifier_name rule. There would be benefits to catch names that are `id` outside of Identifiable types that should still flag as a warning.

4. Should the rule be opt-in or enabled by default? Why?

Enabled by default, it is part of the SwiftUI framework. Unlikely that Apple will break their protocol at this point and rename `id` to `identifier`.

**Comment by 72A12F4E** [comment]
I've looked into what it might take to make this work. I'm stuck at the point where I need to get the enclosing type's information for the variable declaration being linted, but I'm not sure how to really do that.

My branch is here if anyone wants to take a look. 

https://github.com/72A12F4E/SwiftLint/tree/fix_identifier_name_identifiable

**Comment by agarmash** [comment]
I believe it's a good idea to exclude `id` from this rule by default since it's a well-established term in programming
